### PR TITLE
Using dummy data to render home view + reshuffle

### PR DESCRIPTION
**Pull request by wilsonpage**

### CODE REVIEW

**Comment by wilsonpage** [comment]
FYI I rebased on top of your Redux Placeholder PR (#5) to get that goodness involved.

**Comment by wilsonpage** [comment]
@gmarty r?

File: js/App.js [hunk 1]
```diff
@@ -108,4 +111,9 @@ const styles = StyleSheet.create({
     flex: 1,
     backgroundColor: theme.colorBackground,
   },
+
+  scene: {
+    backgroundColor: '#333',
+    paddingTop:  Platform.OS === 'ios' ? 20 : 24,
```
Comment (gmarty) [inline js/App.js:119]: Nice!

File: js/AppProvider.js [hunk 2]
```diff
@@ -1,9 +1,19 @@
 import React, { Component } from 'react';
 import { Provider } from 'react-redux';
-import store from './store';
+import createStore  from './store';
```
Comment (gmarty) [inline js/AppProvider.js]: Let's stick to React best practices of naming files after their default export.
So `./store` should be named `./store/createStore` or keep the import as `store`.

File: js/AppProvider.js [hunk 3]
```diff
@@ -1,9 +1,19 @@
...
 
 /**
+ * Create the global app redux store.
+ *
+ * This store is a singleton that lasts
+ * for the lifetime of the app.
+ *
+ * @type {ReduxStore}
+ */
+const store = createStore();
```
Comment (gmarty) [inline js/AppProvider.js:14]: What's the benefit of calling `createStore()` rather than from the store file?

File: ios/MagnetJourney.xcodeproj/project.pbxproj [hunk 4]
```diff
@@ -711,7 +711,7 @@
 					"$(PROJECT_DIR)",
 				);
 				OTHER_LDFLAGS = "$(inherited)";
-				PRODUCT_BUNDLE_IDENTIFIER = org.reactjs.native.example.journey;
+				PRODUCT_BUNDLE_IDENTIFIER = me.wilsonpage.journey;
```
Comment (samgiles) [inline ios/MagnetJourney.xcodeproj/project.pbxproj]: Are these changes intentional?

**Review by samgiles — COMMENTED** [review]
Didn't mean to "start a review"..

File: js/components/Header.js [hunk 5]
```diff
@@ -0,0 +1,121 @@
...
+    fontSize: 14,
+    opacity: 0.7,
+  },
+
+  titleText: {
+    color: 'white',
+    fontSize: 19,
+    marginTop: -2,
+  },
+});
```
Comment (gmarty) [inline js/components/Header.js:125]: It's unfortunate to remove the underlaying native [ToolbarAndroid element](https://facebook.github.io/react-native/releases/0.40/docs/toolbarandroid.html) in Android.
It comes with all placeholders and styling already in place for navigation.
The styling is then only required on iOS.

File: js/components/Header.js [hunk 6]
```diff
@@ -0,0 +1,121 @@
...
+  }
+
+  renderBackButton() {
+    if (!this.canGoBack()) return;
+
+    return <TouchableOpacity
+      style={styles.action}
+      onPress={this.onBackPress}>
+      <Text style={styles.actionText}>Back</Text>
+    </TouchableOpacity>;
```
Comment (gmarty) [inline js/components/Header.js]: nit: wrap in parens.

File: js/scenes/Home.js [hunk 7]
```diff
@@ -0,0 +1,155 @@
...
+} from '../store/constants';
+
+export class Home extends Component {
+  constructor(props) {
+    super(props);
+
+    this.navigator = this.props.navigator;
+
+    // never bind functions in render(), it
+    // messes with react's diffing logic
```
Comment (gmarty) [inline js/scenes/Home.js:30]: Do we need to repeat this in every component?

File: js/scenes/Home.js [hunk 8]
```diff
@@ -0,0 +1,155 @@
...
+    switch (status) {
+      case EMPTY: return;
+      case FETCHING: return this.renderItemsFetching();
+      case ERRORED: return this.renderItemsErrored();
+      case FETCHED: return this.renderList();
+    }
+  }
+
+  renderItemsFetching() {
+    return <ActivityIndicator
```
Comment (gmarty) [inline js/scenes/Home.js]: nit: wrap in parens.
Same for all methods below.

File: js/scenes/Map.js [hunk 9]
```diff
@@ -23,21 +19,19 @@ export default class Map extends Component {
...
-        }}
-      >
-        <Header title="Map"/>
-
-        <TouchableOpacity
-          onPress={this.onBackPress}>
-          <Text style={styles.text}>Back</Text>
-        </TouchableOpacity>
-      </MagnetMapView>
+      <View style={{ flex: 1 }}>
```
Comment (gmarty) [inline js/scenes/Map.js]: Move styling to its own `styles` constant.

File: js/scenes/Settings.js [hunk 10]
```diff
@@ -46,8 +49,8 @@ export default class Settings extends Component {
         ))}
 
         <TouchableOpacity
-          onPress={this.onBackPress}>
-          <Text style={styles.text}>Back</Text>
+          onPress={() => this.navigator.push({ id: 'demos' })}>
```
Comment (gmarty) [inline js/scenes/Settings.js]: Remove the code related to `onBackPress` if unused.

File: js/api/fetch-items/index.js [hunk 11]
```diff
@@ -0,0 +1,10 @@
+
```
Comment (gmarty) [inline js/api/fetch-items/index.js]: nit: no line break at the beginning of files.

File: js/store/actions.js [hunk 12]
```diff
@@ -0,0 +1,73 @@
+'use strict';
```
Comment (gmarty) [inline js/store/actions.js]: Strict mode is activated by default on JS modules, so when using the `import` this directive is not required anymore.

File: js/scenes/Home.js [hunk 13]
```diff
@@ -0,0 +1,120 @@
+
```
Comment (gmarty) [inline js/scenes/Home.js]: nit: no line break at the beginning of files.

File: ios/MagnetJourney/Info.plist [hunk 14]
```diff
@@ -22,24 +22,7 @@
 	<string>1</string>
 	<key>LSRequiresIPhoneOS</key>
 	<true/>
-	<key>UILaunchStoryboardName</key>
```
Comment (arcturus) [inline ios/MagnetJourney/Info.plist]: Why is this file changed?

File: js/store/reducers.js [hunk 15]
```diff
@@ -0,0 +1,44 @@
+
```
Comment (gmarty) [inline js/store/reducers.js]: nit: no line break at the beginning of files.

File: ios/MagnetJourney.xcodeproj/project.pbxproj (hunk 4 above)
Comment (arcturus) [reply ios/MagnetJourney.xcodeproj/project.pbxproj]: This change shouldn't be committed.

**Review by gmarty — CHANGES_REQUESTED** [review]
There are plenty of code formatting that will be fixed by running `yarn run fix`, so please run it before committing. We're going to make it precommit hook.

Also, please split your PR into meaningful set of changes. There are 3 different unrelated changes made here. That makes the review harder and pollute the changelog and the features depending on it, like `git blame`.

I left some comments and a few nits.

**Comment by arcturus** [comment]
split this pr :), everyone will be happier

**Comment by arcturus** [comment]
Looking gorgeous btw, should we ship just this? ;)

File: js/AppProvider.js (hunk 3 above)
Comment (wilsonpage) [reply js/AppProvider.js:14]: Having `createStore()` factory allows us to create stores for unit-testing. Prior to this we only had access to a singleton.

File: js/components/Header.js (hunk 5 above)
Comment (wilsonpage) [reply js/components/Header.js:125]: I understand. But it became more work to maintain two components. The header needed a back button and the ability to add actions (eg. 'settings')

File: js/store/actions.js (hunk 12 above)
Comment (wilsonpage) [reply js/store/actions.js]: Cool never knew that, thanks :)

**Comment by wilsonpage** [comment]
@gmarty getting error running `yarn run fix`

```bash
~/Documents/m/mozilla-magnet/journey (10-add-basic-home-view) yarn run fix
yarn run v0.15.1
$ "eslint ./*.js ./js/ ./__test__/ --fix" 
sh: eslint ./*.js ./js/ ./__test__/ --fix: No such file or directory
error Command failed with exit code 127.
info Visit http://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

File: js/AppProvider.js (hunk 3 above)
Comment (gmarty) [reply js/AppProvider.js:14]: That makes sense.

File: js/components/Header.js (hunk 5 above)
Comment (gmarty) [reply js/components/Header.js:125]: Sounds good.

**Comment by gmarty** [comment]
I may have eslint installed globally. Can you try:
```bash
node node_modules/.bin/eslint ./*.js ./js/ ./__test__/ --fix
```

**Comment by arcturus** [comment]
@wilsonpage can you rebase?

**Comment by arcturus** [comment]
lgtm after the changes suggested by @gmarty, please fix the navigation in the new compass scene. You need to pass the navigator to the <header> component in the compass scene.

🚢 🚢 🚢 🚢 🚢 🚢 🚢 🚢 🚢
